### ISSUE TITLE
Documentation for development

### ISSUE THREAD

**Issue by linuswagner** [usethesource/rascal-language-servers#282]
It would be helpful to have a description of how to develop on this project including:

- how to get the project to compile and run?
- how do different module interact with each other -> when do I need to touch what? what do I need to have checkout/change if I want to add a certain feature?

At the moment, I am trying to set up the project by opening `rascal-vscode-extension` in VS Code, run `npm install`, and run the extension through the run menu. For some reason, I am unable to run the extension without any modification and the console shows a random error. I feel like this is more an issue of my lack of knowledge on how to build and run the project rather than an issue of the code.

**Comment by DavyLandman** [comment]
Agreed. We should have a developer section in the readme. Maybe even think about setting up a devcontainers configuration?

Specifically for your problem: you have to run `./build.sh -f` at least once so that the maven is run and generates the jars that the vs code extension starts. And when you are launching vs code in 2nd level, you'll also need to launch rascal-lsp server by hand.

**Comment by jurgenvinju** [comment]
Let's have a developer section in a separate .md tutor file. The readme can point to it. There are going to be different files for different topics regarding development.

**Comment by DavyLandman** [comment]
What is wrong with having a small section in the readme that says: for developers of this project (so not it's users), this is a development setup you should use? It's quite a common pattern to have that in the readme?
